### PR TITLE
show osm user profile page

### LINKED ISSUE #1093: Update profile page
# Problem
Currently we have profile page for our healthsite user, but we are using osm data that the uploader is not there

# Solution 
Update profile page to use user regardless healthsite user

### PR DESCRIPTION
**Pull request by ann26**
fix #1093 

Screenshot from 2019-04-10 17-16-25

### CODE REVIEW

File: django_project/social_users/views.py [hunk 1]
```diff
@@ -97,39 +110,57 @@ def save_profile(backend, user, response, *args, **kwargs):
...
-        Locality.objects
-        .filter(changeset__social_user=user).exclude(changeset__in=changesets)
-        .order_by('-changeset__created')
-        .values('changeset', 'changeset__created', 'changeset__social_user__username', 'version')
-        .annotate(edit_count=Count('changeset'), locality_id=Max('id'))[:10]
-    )
-    for update in updates_temp:
-        updates.append(get_update_detail(update))
+
+    if not osm_user:
```
Comment (meomancer) [inline django_project/social_users/views.py]: i think we don't need this section, all of user is using osm data (use HS username to check on docker osm data)

File: django_project/social_users/views.py [hunk 2]
```diff
@@ -97,39 +110,57 @@ def save_profile(backend, user, response, *args, **kwargs):
...
+        # get from locality osm for osm users
+        updates_osm = (
+            LocalityOSMView.objects
+            .filter(changeset_user=user)
+            .order_by('-changeset_timestamp')
+            .values('changeset_id', 'changeset_timestamp', 'changeset_user', 'changeset_version', 'name')
+            .annotate(edit_count=Count('changeset_id'))
+        )
+
+        for update in updates_osm:
```
Comment (meomancer) [inline django_project/social_users/views.py]: please create serializer foe changets of locality osm

File: django_project/localities/utils.py [hunk 3]
```diff
@@ -57,7 +57,10 @@ def extract_updates(updates):
                 update['locality'] = locality.name
             except Locality.DoesNotExist:
                 update['locality_uuid'] = 'unknown'
-                update['locality'] = 'unknown'
+                if not update['name']:
```
Comment (meomancer) [inline django_project/localities/utils.py]: We don't need this changes anymore

File: django_project/localities_osm/serializer/locality_osm.py [hunk 4]
```diff
@@ -42,3 +44,28 @@ class Meta:
...
+            'changeset_id',
+            'changeset_version',
+            'changeset_timestamp',
+            'changeset_user',
+        ]
+
+    def to_representation(self, instance):
+        result = super(LocalityOSMUpdates, self).to_representation(instance)
+        result['changeset__social_user__username'] = None
+        print(result['changeset_timestamp'])
```
Comment (meomancer) [inline django_project/localities_osm/serializer/locality_osm.py]: remove this print

File: django_project/localities_osm/serializer/locality_osm.py [hunk 5]
```diff
@@ -42,3 +44,28 @@ class Meta:
...
+            'name',
+            'changeset_id',
+            'changeset_version',
+            'changeset_timestamp',
+            'changeset_user',
+        ]
+
+    def to_representation(self, instance):
+        result = super(LocalityOSMUpdates, self).to_representation(instance)
+        result['changeset__social_user__username'] = None
```
Comment (meomancer) [inline django_project/localities_osm/serializer/locality_osm.py]: could you use field serializer so we have nice code?

Comment (ann26) [reply django_project/localities_osm/serializer/locality_osm.py]: ok @meomancer

**Comment by timlinux** [comment]
Nice thanks @ann26 !

**Comment by meomancer** [comment]
thank you @ann26 
great job
i merged this PR (already tested and works well)

**Pull request merged**